### PR TITLE
Enable headless linux build tests

### PR DESCRIPTION
**Pull request by mattseddon**
This PR was promised here: https://github.com/ryanluker/vscode-coverage-gutters/pull/314#issuecomment-851379606

It enables tests against a linux build in the CI and also adds commands and packages that make running the test suite from the command line inside of the devcontainer possible.

I tried to make everything available in the most elegant way possible but github actions always end up being ugly (at best). LMK what you think and if you would like to see anything changed.

Thanks @ryanluker 👍🏻 

Note: If you accept these changes then the required / expected steps will needed to be updated under the repository settings (the way that the actions now work has changed the matrix names, there are other ways to solve that problem but this was my preference).

### CODE REVIEW

File: .devcontainer/Dockerfile [hunk 1]
```diff
@@ -3,12 +3,13 @@ ARG VARIANT="14"
 FROM mcr.microsoft.com/vscode/devcontainers/javascript-node:0-${VARIANT}
 
 # [Optional] Uncomment this section to install additional OS packages.
-# RUN apt-get update && export DEBIAN_FRONTEND=noninteractive \
-#     && apt-get -y install --no-install-recommends <your-package-list-here>
+RUN apt-get update && export DEBIAN_FRONTEND=noninteractive \
+	&& apt update \ 
+	&& apt install -y libasound2 libgbm-dev libgtk-3-0 libnss3 xvfb
```
Comment (mattseddon) [inline .devcontainer/Dockerfile:8]: [F] These are the extra packages that are needed to make `xvfb-run` work inside the devcontainer.

File: package.json [hunk 2]
```diff
@@ -247,8 +247,11 @@
     "pretty": "tslint --fix './src/**/*.ts' './test/**/*.ts'",
     "pre-commit": "lint-staged",
     "test": "npm run compile && node ./out/test/runTest.js",
+    "test-headless": "xvfb-run -a npm run test",
     "test-unit": "cross-env MOCHA_GREP=@unit npm test --silent",
+    "test-unit-headless": "xvfb-run -a npm run test-unit",
     "test-integration": "cross-env MOCHA_GREP=@integration npm test --silent",
+    "test-integration-headless": "xvfb-run -a npm run test-integration",
```
Comment (mattseddon) [inline package.json:254]: [F[ These headless options means that the tests can run from the command line in the devcontainer and in the CI via for linux via the same command.

File: .devcontainer/Dockerfile (hunk 1 above)
Comment (ryanluker) [reply .devcontainer/Dockerfile:8]: gotcha makes sense!

File: package.json (hunk 2 above)
Comment (ryanluker) [reply package.json:254]: excellent this makes a lot of sense especially what we saw in the previous PR where we discussed this and the documentation provided my microsoft.
https://code.visualstudio.com/api/working-with-extensions/continuous-integration#github-actions

**Review by ryanluker — APPROVED** [review]
Thanks again for the contribution!
I will reactivate the requirement that PRs have the linux job passing once we merge this in.

**Comment by ryanluker** [comment]
@mattseddon I have setup the new branch requirements correctly I believe 🤔 I am planning to work on a ticket today so we will see!
I assume this is what you meant by "repository settings".
image

**Comment by mattseddon** [comment]
> @mattseddon I have setup the new branch requirements correctly I believe 🤔.
> I assume this is what you meant by "repository settings".
> 
> I am planning to work on a ticket today so we will see!
> image
> 
> I also saw this linux option for the branch protection but I believe we want to use the headless option to get your improvements correct?
> image

The options that you have set look right. It will be pretty obvious when you make a PR if they are not 👍🏻.

**Comment by ryanluker** [comment]
@mattseddon excellent thanks again for the contributions they help a ton!
happy code coverages 😁.

**Comment by mattseddon** [comment]
> @mattseddon excellent thanks again for the contributions they help a ton!
> happy code coverages 😁.

No worries, thank you for the extension. I've been using it for a long time now! Let me know which issue(s) you're working on and I'll try to pick off a different one over the next couple of week 👍🏻.

**Comment by ryanluker** [comment]
@mattseddon Thanks! the issues backlog is pretty nicely laid out, see if any would be something you would want to work on 🤔.
 usually assign myself to ones I am actively thinking about or working on (currently https://github.com/ryanluker/vscode-coverage-gutters/issues/308 ).

**Pull request merged**